### DISCUSSION TITLE
`from_python` and `from_cpp` must be marked as `noexcept`?

### DISCUSSION THREAD

**Discussion by zhqrbitee** [wjakob/nanobind#272] [Q&A]
Hi,
Based on the docs here: https://nanobind.readthedocs.io/en/latest/porting.html#type-casters, `from_python` and `from_cpp` must be marked as `noexcept`.   
1. So my question is why? It just feel strange to me to mark some complicate conversion logic as `noexcept`. 
2. What would happen if I didn't mark them as `noexcept`?

**Comment by wjakob** [comment]
If you raise exceptions from a type caster missing the `noexcept` annotation, it will break dispatch of functions with multiple overloads.

See https://github.com/wjakob/nanobind/commit/b3c3723a2db2aae62a18e330f24b1fafe097cf2b for the rationale.
